apps/judicial-system/backend/src/app/modules/indictment-count/guards/offenseExists.guard.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  BadRequestException,
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
+
+import { Case } from '../../case'
+
+@Injectable()
+export class OffenseExistsGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest()
+    const theCase: Case = request.case
+    const indictmentCountId = request.params.indictmentCountId
+    const indictmentCount = theCase.indictmentCounts?.find(
+      (indictmentCount) => indictmentCount.id === indictmentCountId,
+    )
```

Comment:
```suggestion
    const indictmentCount = theCase.indictmentCount
    
    if (!indictmentCount) {
      throw BadRequestException('Missing indictment count')
    }
```

Reply:
So, the idea here is to assume that the request constains everything that the offense exists guard needs. In this case, it needs an indictment count from the request. The above code suggestion is what I would expect to see.

Please consider applying this change 😃 Some tests may need updates.

Reply:
Ah I thought I had changed this already 🤔 Will fix 💪